### PR TITLE
Add PR and Issue template

### PR DESCRIPTION
**Pull request by canavandl**
ref #4015

Add PR and issue template for Github.
* Issue template will stress copy/pastable code example, stack traces and software versions.
* PR template will note need for related issue, tests, release notes entry.

### CODE REVIEW

**Comment by bryevdv** [comment]
png @bokeh/dev any comments on the wording of these? They will "pre-populate" the GH issue and PR submission forms.

**Comment by dennisobrien** [comment]
For the issue template, it might be helpful to add sections for "expected behavior" and "observed behavior".  A good bug description would contain this information anyway, but spelling it out can help bug reporters be more explicit.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Description of defect, deficiency or desired feature.
+
+For defects or deficiencies:
```
Comment (almarklein) [inline .github/ISSUE_TEMPLATE.md]: suggestion: "... deficiencies, please provide:"

**Comment by canavandl** [comment]
@dennisobrien @almarklein 

Thanks for the feedback. I included the suggestions.

**Comment by bryevdv** [comment]
Suggest adding a loud disclaimer on issues that GH is not the place for general support questions (with a link to the mailing list)

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+For questions and technical assistance, come ask the [Bokeh mailing list](https://groups.google.com/a/continuum.io/forum/#!forum/bokeh).
```
Comment (canavandl) [inline .github/ISSUE_TEMPLATE.md]: @bryevdv do you think this is clear enough? I could also add a "The issue tracker isn't the place for general support. For questions..." or similar, but I don't want to sound mean.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+All pull requests must have an associated issue in issue tracker. If there
```
Comment (almarklein) [inline .github/PULL_REQUEST_TEMPLATE.md]: "in THE issue tracker", or does this just sound strange to me because I am Dutch? :)

Comment (canavandl) [inline .github/PULL_REQUEST_TEMPLATE.md]: Nope - that's a typo.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+The issue tracker isn't the place for general support. For questions and
+technical assistance, come ask the
+[Bokeh mailing list](https://groups.google.com/a/continuum.io/forum/#!forum/bokeh).
+
```
Comment (damianavila) [inline .github/ISSUE_TEMPLATE.md]: Maybe we should add the gitter chat reference as well??

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+For defects or deficiencies, please provide:
+* Description of expected behavior and the observed behavior
+* self-contained example code that generates the issue (if possible)
+* stack traceback and/or browser console error output
```
Comment (damianavila) [inline .github/ISSUE_TEMPLATE.md]: we probably need a link to let know the user how to look into the js console... some users does not know about that...

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+* python, bokeh and browser versions
```
Comment (damianavila) [inline .github/ISSUE_TEMPLATE.md]: we maybe need to reference the print versions thing we have: https://github.com/bokeh/bokeh/blob/84423bac8a9897236e8e84f22492bd77760f10ea/bokeh/util/testing.py#L23

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+All pull requests must have an associated issue in the issue tracker. If there
```
Comment (damianavila) [inline .github/PULL_REQUEST_TEMPLATE.md]: Something like, you can read about our issues and PR processes here... and link the corresponding BEP.

File: .github/ISSUE_TEMPLATE.md (hunk 6 above)
Comment (bryevdv) [inline .github/ISSUE_TEMPLATE.md]: Much easier now to just run `bokeh info`
```
$ bokeh info
Python version      :  2.7.11 |Continuum Analytics, Inc.| (default, Dec  6 2015, 18:57:58) 
IPython version     :  Not installed
Bokeh version       :  0.11.1
BokehJS static path :  /Users/bryan/anaconda/envs/py27/lib/python2.7/site-packages/bokeh/server/static
```

Comment (damianavila) [inline .github/ISSUE_TEMPLATE.md]: yep... better!!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (canavandl) [inline .github/PULL_REQUEST_TEMPLATE.md]: Added

**Comment by canavandl** [comment]
Current issue template loaded:

screen shot 2016-03-18 at 1 49 36 pm

Is it too long? The links are unrendered markdown - should they be laid out differently?

**Comment by canavandl** [comment]
Proposed PR template

screen shot 2016-03-18 at 2 02 31 pm

screen shot 2016-03-18 at 2 04 05 pm

**Comment by bryevdv** [comment]
It is a bit long, I'd just mention the  ML, not gitter, and maybe not bother with the extra markup for the URL

**Comment by canavandl** [comment]
@bryevdv do you mind if we leave the Gitter reference and url markup in? I'm curious to see how many people show up on the Gitter (I think it'll be the best way to seed the channel) and I like the clickable url from preview mode.

I agree it's the issue template is a little long, but I expect to revisit it in a couple of weeks based on feedback/quality of new issues that are opened.

**Comment by bryevdv** [comment]
OK

**Comment by damianavila** [comment]
LGTM

**Comment by bryevdv** [comment]
OK let's see how it goes.

**Pull request merged**